affinity/cpp-20/d0796r1.md

```diff
@@ -255,6 +255,10 @@ An interesting question which arises here is whether the system topology of an e
 
 Note that this is different from devices that go online or offline during execution: The devices themselves are online, they have not been found (or used) by the program until the appropriate discovery stage has been executed.
 
+There are generally two main reasons why you would want the **system resource topology** to be dynamic after the initial point of discovery. Firstly for fault tolerance, being able to handle a resource becoming unavailable at some point after the initial discovery while the application is running. Secondly for detecting a new resource, being able to discover new resources added to the system at some point after the initial discovery while the application is running. Though the former case is much more common than the latter.
```

Comment:
Does "fault tolerance" imply that parallel execution may fail, for example if someone unplugs a USB compute device?  It seems like we're moving towards a model where the "main thread" may have a different fault model than executors.  I like that but parallel fault tolerance is hard (MPI has been debating this forever) so we may want to be cautious about dipping our toes into it.

Reply:
I am not keen of adding fault tolerance at all at this point, since as you suggest, it can get quite complicated. This explanation comes from some questions that asked if that was considered and what would it entail, and we are basically saying "yeah its there there is experience but we will not address it on this proposal". Maybe is not clear and we need some extra statements...

Reply:
@Ruyk I think it's OK to allude to the possibly of fault tolerance, but I think the last sentence in your comment "Being able to query for resource availability ... enables reliable programming to be part of the core language" goes too far.

Reply:
Agree, I've softened the statement above

Reply:
@Ruyk @mhoemmen I've revised the wording to include your suggestions. I also agree that although it's important to highlight the possibility of fault tolerance it should be left out of the scope of this paper.